admin/admin.py

```diff
@@ -0,0 +1,94 @@
+from flask import Flask
+from flask_basicauth import BasicAuth
+from flask_admin import Admin
+from flask_sqlalchemy import SQLAlchemy
+from flask_admin.contrib.sqla import ModelView
+from sqlalchemy import Table
+import configparser
+import argparse
+import sys
+import os
+
+
+def get_config_ini(config_file="../conf/harvester.conf"):
+    '''
+    Read ini-formatted config file from disk
+    :param config_file: Filename of config file
+    :return: configparser-style config file
+    '''
+
+    config = configparser.ConfigParser()
+    config.read(config_file)
+    return config
+
+
+def run_admin_server(with_tls=False):
+    '''
+    Read ini-formatted config file from disk
+    :param with_tls: Boolean on whether to run with TLS security or not.
+    '''
+
+    # Initial setup magic
+    config = get_config_ini()
+    app = Flask(__name__)
+    basic_auth = BasicAuth(app)
+    db_connection_str = "postgresql://" + str(config['db'].get('user')) + ":" + str(config['db'].get('pass')) + "@" \
```

Comment:
Technically every Python linter hates making big strings by just endlessly concatting `str() + str()` but I still do everything that way so I'm the last person to object, just FYI